pkg/stanza/fileconsumer/matcher/internal/finder/finder.go

```diff
@@ -20,10 +21,15 @@ func Validate(globs []string) error {
 }
 
 // FindFiles gets a list of paths given an array of glob patterns to include and exclude
-func FindFiles(includes []string, excludes []string) []string {
+func FindFiles(includes []string, excludes []string) ([]string, error) {
+	var errs error
 	all := make([]string, 0, len(includes))
 	for _, include := range includes {
-		matches, _ := doublestar.FilepathGlob(include, doublestar.WithFilesOnly()) // compile error checked in build
+		matches, err := doublestar.FilepathGlob(include, doublestar.WithFilesOnly(), doublestar.WithFailOnIOErrors())
+		if err != nil {
+			errs = errors.Join(errs, fmt.Errorf("find files with '%s' pattern: %w", include, err))
+			matches, _ = doublestar.FilepathGlob(include, doublestar.WithFilesOnly())
```

Comment:
Just making sure I understand correctly:  The reason to do this is because the same pattern could cause an IO error due to one file or directory, but also could still find files if we do not provide the option. 

If this is right, can we just have a comment here to explain this? It can be very brief but I think it's worth noting so that future refactoring does not miss this nuance.

Reply:
I see the test case you added shows this quite nicely. Still I think a comment is worthwhile.